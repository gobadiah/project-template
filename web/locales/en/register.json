{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
