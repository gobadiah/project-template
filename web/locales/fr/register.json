{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
