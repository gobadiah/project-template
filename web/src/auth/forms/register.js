import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from '~/components/forms/fields';
import { Form } from '~/components/forms';

import { PureComponent } from '~/components/base';

class RegisterForm extends PureComponent {
  render() {
    return (
      <Form {...this.props}>
        <EmailField key='email' />
        <PasswordField
          key='password'
          autoComplete='new-password'
        />
        <PasswordField
          key='confirm_password'
          name='confirm_password'
          label='Password confirmation'
          autoComplete='new-password'
        />
        <FirstNameField key='first_name' />
        <LastNameField key='last_name' />
        <BirthdayField key='birthday' />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
