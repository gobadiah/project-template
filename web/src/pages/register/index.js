import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
