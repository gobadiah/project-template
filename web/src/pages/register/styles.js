import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
