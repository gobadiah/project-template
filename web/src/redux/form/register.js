import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
