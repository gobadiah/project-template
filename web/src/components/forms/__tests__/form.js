import 'jsdom-global/register';

import React from 'react';

import { mount, shallow } from 'enzyme';

import { Error, Form, Warning } from '..';

const props = {
  children: <div />,
  error: 'One error',
  handleSubmit: () => {},
  pristine: false,
  submitting: false,
  reset: () => {},
  warning: 'One warning',
};

const context = {
  i18n: {
    on: jest.fn(),
  },
  t: x => x,
};

describe('Form', () => {
  it('should match snapshot when mounting', () => {
    const tree = mount(<Form {...props} />, { context });
    expect(tree).toMatchSnapshot();
  });

  it('shallow render should contain elements', () => {
    const wrapper = shallow(<Form {...props} />, { context });
    expect(wrapper.contains(<Error error={props.error} />)).toBeTruthy();
    expect(wrapper.contains(<Warning warning={props.warning} />)).toBeTruthy();
  });
});
