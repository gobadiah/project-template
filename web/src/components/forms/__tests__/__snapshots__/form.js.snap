// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot when mounting 1`] = `
<Form
  error="One error"
  handleSubmit={[Function]}
  pristine={false}
  reset={[Function]}
  submitting={false}
  warning="One warning"
>
  <form
    onSubmit={[Function]}
  >
    <div />
    <input
      disabled={false}
      type="submit"
      value="Send"
    />
    <input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Reset"
    />
    <Warning
      warning="One warning"
    >
      <div>
        One warning
      </div>
    </Warning>
    <Error
      error="One error"
    >
      <div>
        One error
      </div>
    </Error>
  </form>
</Form>
`;
